### DISCUSSION TITLE
Which of the given themes takes less time to load

### DISCUSSION THREAD

**Discussion by Atharv115** [JanDeDobbeleer/oh-my-posh#2548] [Troubleshoot]
Hi, Guys 

I wanted to know which theme takes less time to load
![image](https://user-images.githubusercontent.com/86460457/179989525-0c8dad39-4680-4570-bdea-f52dd55ca013.png)

I'm newto Discussions, If I've made any mistake in asking this question Please do forgive me
Thanks

**Comment by lewis-yeung** [comment]
@Atharv115 You can run `oh-my-posh debug -c <path-to-a-theme-config>` to see how much time oh-my-posh takes to render a prompt. You'll see a line of `Run duration: ***ms`.

In addition, if the environment variable `POSH_THEMES_PATH` has been set (it is automatically set if you install oh-my-posh via WinGet/Scoop on Windows), you can run the command below in PowerShell to test all the theme configs:

```powershell
Get-ChildItem -Path "$env:POSH_THEMES_PATH\*.omp.*" | ForEach-Object { oh-my-posh debug -c "$_" } >test.txt
```

This way you can search `Run duration:` in the output file `test.txt` to find out which takes the least time.

BTW, currently `oh-my-posh debug` may hang on some theme configs, this is likely a bug. See #2549.

**Reply by lewis-yeung** [reply]
Uh-oh, I forgot the most important point. The load time which you're really concerned about doesn't include the time that OMP takes to render a prompt. For OMP, only the command `oh-my-posh init pwsh ...` is executed during loading the PowerShell profile.

**Reply by Atharv115** [reply]
> Uh-oh, I forgot the most important point. The load time which you're really concerned about doesn't include the time that OMP takes to render a prompt. For OMP, only the command `oh-my-posh init pwsh ...` is executed during loading the PowerShell profile.

I didn't quite get what you mean, like **The load time which you're really concerned about doesn't include the time that OMP takes to render a prompt. For OMP, only the command `oh-my-posh init pwsh ...` is executed during loading the PowerShell profile.** Would u plz elaborate on it coz I am _**completely naive**_

**Comment by JanDeDobbeleer** [comment]
@Atharv115 the load time is a PowerShell core issue, not oh-my-posh. You can validate the difference when using PowerShell 5, or another shell.

**Reply by Atharv115** [reply]
fine @JanDeDobbeleer, but is there any way that it can be reduced?

**Reply by JanDeDobbeleer** [reply]
not that I know, you need to pick that up with the PowerShell team.

**Reply by Atharv115** [reply]
so basically no chance, ryt @JanDeDobbeleer
